### PR TITLE
Some small sorries

### PR DESCRIPTION
**Pull request by michaelmcinerney**
This also includes `corres_split_skip`, a little rule to skip over a function in a `corres` proof in a way similar to `hoare_seq_ext_skip`

### CODE REVIEW

File: proof/refine/ARM/Tcb_R.thy [hunk 1]
```diff
@@ -485,45 +485,45 @@ lemma arch_post_modify_registers_corres:
...
 
+crunches Tcb_A.restart
+  for ex_nonz_cap_to[wp]: "ex_nonz_cap_to tcb_ptr"
+  (wp: crunch_wps cancel_ipc_cap_to simp: crunch_simps)
+
+crunches restart
+  for ex_nonz_cap_to'[wp]: "ex_nonz_cap_to' tcbPtr"
+  (wp: crunch_wps threadSet_cap_to simp: crunch_simps tcb_cte_cases_def)
+
+(* FIXME RT: move? *)
```
Comment (MitchellBuckley) [inline proof/refine/ARM/Tcb_R.thy:496]: Yes, move before merging. I like it!

**Review by MitchellBuckley — APPROVED** [review]
Excellent!!

File: proof/refine/ARM/Interrupt_R.thy [hunk 2]
```diff
@@ -649,16 +651,23 @@ lemma tcbSchedAppend_invs_but_ct_not_inQ':
...
              valid_irq_handlers_lift' cur_tcb_lift ct_idle_or_in_cur_domain'_lift2
              untyped_ranges_zero_lift
         | simp add: cteCaps_of_def o_def
         | fastforce elim!: st_tcb_ex_cap'' split: thread_state.split_asm)+
-  done *)
+  done
 
 lemmas corres_eq_trivial = corres_Id[where f = h and g = h for h, simplified]
 
+lemma doMachineOpackDeadlineIRQ_invs'[wp]:
```
Comment (corlewis) [inline proof/refine/ARM/Interrupt_R.thy:663]: Maybe
```suggestion
lemma doMachineOp_ackDeadlineIRQ_invs'[wp]:
```

File: proof/refine/ARM/Tcb_R.thy [hunk 3]
```diff
@@ -485,45 +485,45 @@ lemma arch_post_modify_registers_corres:
   apply (rule corres_stateAssert_assume)
    by simp+
 
+crunches Tcb_A.restart
```
Comment (corlewis) [inline proof/refine/ARM/Tcb_R.thy:488]: Can you stick a FIXME move on this one as well? Or maybe move it to Move.thy

**Review by corlewis — APPROVED** [review]
Nice work!

File: proof/refine/ARM/Interrupt_R.thy [hunk 4]
```diff
@@ -356,8 +356,8 @@ lemma invoke_arch_irq_handler_corres:
 
 lemma invoke_irq_handler_corres:
   "irq_handler_inv_relation i i' \<Longrightarrow>
-   corres dc (einvs and irq_handler_inv_valid i)
-             (invs' and irq_handler_inv_valid' i')
+   corres dc (einvs and irq_handler_inv_valid i and simple_sched_action)
+             (invs' and irq_handler_inv_valid' i' and sch_act_simple)
```
Comment (mktnk3) [inline proof/refine/ARM/Interrupt_R.thy:360]: This is not about this PR, but we definitely can be more consistent in naming these concepts... 😆

File: proof/refine/ARM/Interrupt_R.thy [hunk 5]
```diff
@@ -699,17 +707,9 @@ lemma handle_interrupt_corres:
...
-        apply (rule corres_eq_trivial, simp+)
-         apply (rule corres_machine_op)
-         apply (rule corres_eq_trivial, (simp add: no_fail_ackInterrupt)+)
-        apply wp+
-   apply clarsimp
-  apply clarsimp
-  done *)
+  apply (corressimp corres: corres_machine_op reprogram_timer_corres
+                      simp: ackDeadlineIRQ_def)
+  done
```
Comment (mktnk3) [inline proof/refine/ARM/Interrupt_R.thy:712]: Nice

**Review by mktnk3 — APPROVED** [review]
Nice. I’ll see and fix up the bamboo sorry count for Refine.

**Comment by mktnk3** [comment]
> I’ll see and fix up the bamboo sorry count for Refine.
Done

**Pull request merged**